### PR TITLE
JS unit tests new ia

### PR DESCRIPTION
**Pull request by mongomoe**
### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-13431

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/JS-Unit-Tests-New-IA/

### Coverage:
####: Usage Examples:
- [x] Define a Realm Object Schema (https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/JS-Unit-Tests-New-IA/sdk/node/examples/define-a-realm-object-model/) 
- [x] Open and Close a Local Realm (https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/JS-Unit-Tests-New-IA/sdk/node/examples/open-and-close-a-local-realm/)
- [x] Read and Write Data (https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/JS-Unit-Tests-New-IA/sdk/node/examples/read-and-write-data/)
- [x] React to Changes (https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/JS-Unit-Tests-New-IA/sdk/node/examples/react-to-changes/)
- [x] Call a Function (https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/JS-Unit-Tests-New-IA/sdk/node/examples/call-a-function/)

### Note to the reviewer:
- Tests should pass for each page
- Tests should be idempotent and should be able to pass 100% of the time (rather than on only the first run)
- If a realm is opened it should be closed in the same file. Or if it is opened in a test, it should be closed in that test
- if an object is created it should be deleted, and its variable should be nullified

### CODE REVIEW

**Review by nathan-contino-mongo — APPROVED** [review]
LGTM -- examples are equivalent, from what I can tell, and we'll see if they pass CI if you pull in the latest changes in new-ia.

File: examples/node/Examples/read-and-write-data.js [hunk 1]
```diff
@@ -0,0 +1,611 @@
...
+    task = null;
+    task2 = null;
+    task3 = null;
+    task4 = null;
+
+    person1 = null;
+    person2 = null;
+
+    dog1 = null;
+    dog2 = null;
```
Comment (cbush) [inline examples/node/Examples/read-and-write-data.js:295]: I don't think discarding the references is really required. Let's remove that.

File: examples/node/Examples/read-and-write-data.js [hunk 2]
```diff
@@ -0,0 +1,611 @@
...
+    realm.close();
+  });
+  test("should write a new object", async () => {
+    // a realm is opened
+    const realm = await Realm.open({
+      path: "myrealm",
+      schema: [DogSchema, PersonSchema],
+    });
+    // :code-block-start: read-and-write-create-a-new-object
+    // Declare the instance.
```
Comment (cbush) [inline examples/node/Examples/read-and-write-data.js:306]: We could be more precise with our wording here:
```suggestion
    // Declare the variable that will hold the dog instance.
```

File: examples/node/Examples/read-and-write-data.js [hunk 3]
```diff
@@ -0,0 +1,611 @@
...
+    let dog;
+    // Open a transaction.
+    realm.write(() => {
+      // Assign a newly-created instance to the variable.
+      dog = realm.create("Dog", { name: "Max", age: 5 });
+    });
+    // :code-block-end:
```
Comment (cbush) [inline examples/node/Examples/read-and-write-data.js:313]: I might add a line at the end of the code block like `// Use dog` just to show why we bothered with the `let` variable.

File: source/sdk/node/examples/define-a-realm-object-model.txt [hunk 4]
```diff
@@ -60,9 +60,10 @@ object type, define the object schema on the static property ``schema``.
    You use object schemas and class definitions when you :ref:`open a {+realm+}
    <node-open-a-local-realm>` that contains the type.
    
-   .. code-block:: javascript
-      
-      await Realm.open({ schema: [CarSchema] });
+   .. literalinclude:: /examples/generated/node/open-and-close-a-local-realm.codeblock.open-local-realm-with-car-schema.js
+     :language: javascript
+     :emphasize-lines: 4
```
Comment (cbush) [inline source/sdk/node/examples/define-a-realm-object-model.txt:65]: I would go ahead and remove or stop using `emphasize-lines` as there will inevitably be drift between what's in here and what's in the source that we actually want to emphasize. The better option is keeping code examples concise and correct.

**Review by cbush — COMMENTED** [review]
Looks good, but I really think we should add the models used at the top of the page.

File: examples/node/Examples/read-and-write-data.js (hunk 1 above)
Comment (mongomoe) [reply examples/node/Examples/read-and-write-data.js:295]: Without discarding references, the variable is not nulled by default. If you try to console.log(dog1), you would get the error: `Accessing object of type Dog which has been invalidated or deleted`.

Comment (cbush) [reply examples/node/Examples/read-and-write-data.js:295]: I know that, but why are you logging the object in this example? The variables are about to go out of scope. To me, it clutters the code example or unit test.

Comment (cbush) [reply examples/node/Examples/read-and-write-data.js:295]: That said it's only a minor thing.

Comment (mongomoe) [reply examples/node/Examples/read-and-write-data.js:295]: There's actually no logging after nullifying objects on this example itself. It's just a good practice to follow, and if anyone edits this test in the future they won't get any confusing errors. So maintainability + best practice > cluttering of the unit tests in this case imo.

Comment (mongomoe) [reply examples/node/Examples/read-and-write-data.js:295]: Actually, I guess this is a fair point, we probably won't be logging dog after. I'll remove these 'discard the object' bits.

File: examples/node/Examples/read-and-write-data.js (hunk 3 above)
Comment (mongomoe) [reply examples/node/Examples/read-and-write-data.js:313]: 👍

File: source/sdk/node/examples/define-a-realm-object-model.txt (hunk 4 above)
Comment (mongomoe) [reply source/sdk/node/examples/define-a-realm-object-model.txt:65]: Fair point, removed it in this example and will try to keep examples concise in the future 👍

File: examples/node/Examples/read-and-write-data.js (hunk 1 above)
Comment (mongomoe) [reply examples/node/Examples/read-and-write-data.js]: done 👍

**Comment by mongomoe** [comment]
Whoops didn't address your top-level comment. "Looks good, but I really think we should add the models used at the top of the page." 

Added the schemas to the top of the [read-and-write](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/JS-Unit-Tests-New-IA/sdk/node/examples/read-and-write-data/#about-the-examples-on-this-page) page 

@cbush

File: examples/node/Examples/read-and-write-data.js (hunk 1 above)
Comment (cbush) [reply examples/node/Examples/read-and-write-data.js]: Thanks!

File: source/sdk/node/examples/read-and-write-data.txt [hunk 5]
```diff
@@ -12,6 +12,15 @@ Read & Write Data
    :depth: 2
    :class: singlecol
 
+About the Examples on this Page
+-------------------------------
+.. note::
```
Comment (cbush) [inline source/sdk/node/examples/read-and-write-data.txt:17]: Is it possible to remove the note? I feel like it's weird to have a note right after a heading with no other body text.

**Review by cbush — APPROVED** [review]

File: source/sdk/node/examples/read-and-write-data.txt (hunk 5 above)
Comment (mongomoe) [reply source/sdk/node/examples/read-and-write-data.txt]: Fixed it so it's not a note, similar to the iOS page. Thanks 👍

**Pull request merged**